### ISSUE TITLE
Question about the attention mask

### ISSUE THREAD

**Issue by wentaoyuan** [facebookresearch/Mask2Former#125]
May I ask what is the purpose of this line? https://github.com/facebookresearch/Mask2Former/blob/a03be8df3e583a2efbc1139d62b260bf22687259/mask2former/modeling/transformer_decoder/mask2former_transformer_decoder.py#L398 Does it mean that if the mask predicted by the last layer is empty for a specific query, then in the current layer, the query is allowed to attend everywhere, instead of nowhere? This detail is not in the paper. It would be nice to know how important it is to the performance or training convergence.

**Comment by Agito555** [comment]
> May I ask what is the purpose of this line?
> 
> https://github.com/facebookresearch/Mask2Former/blob/a03be8df3e583a2efbc1139d62b260bf22687259/mask2former/modeling/transformer_decoder/mask2former_transformer_decoder.py#L398
> 
> Does it mean that if the mask predicted by the last layer is empty for a specific query, then in the current layer, the query is allowed to attend everywhere, instead of nowhere? This detail is not in the paper. It would be nice to know how important it is to the performance or training convergence.

I think your understanding is right.
